src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -993,12 +1007,41 @@ static ImmutableArray<Symbol> getAllTypeAndRequiredMembers(TypeSymbol containing
                             foreach (var (_, requiredMember) in requiredMembers)
                             {
                                 // We want to assume that all required members were _not_ set by the chained constructor
-                                builder.Add(getFieldSymbolToBeInitialized(requiredMember));
+                                builder.AddRange(getAllMembersToBeDefaulted(requiredMember));
                             }
 
                             return builder.ToImmutableAndFree();
                         }
 
+                        static ImmutableArray<Symbol> getAllMembersToBeDefaulted(Symbol requiredMember)
+                        {
+                            Debug.Assert(requiredMember.IsRequired());
+
+                            if (requiredMember is FieldSymbol)
+                            {
+                                return ImmutableArray.Create<Symbol>(getFieldSymbolToBeInitialized(requiredMember));
+                            }
+                            else
+                            {
+                                var property = (PropertySymbol)requiredMember;
+                                // If the set method is null (ie missing), that's an error, but we'll recover as best we can
+                                var setMethodNotNullMembers = property.SetMethod?.NotNullMembers ?? property.NotNullMembers;
+                                var builder = ArrayBuilder<Symbol>.GetInstance(1 + setMethodNotNullMembers.Length);
+
+                                builder.Add(getFieldSymbolToBeInitialized(property));
+
+                                foreach (var notNullMemberName in setMethodNotNullMembers)
+                                {
+                                    foreach (var member in property.ContainingType.GetMembers(notNullMemberName))
+                                    {
+                                        builder.Add(getFieldSymbolToBeInitialized(member));
```

Comment:
>builder.Add(getFieldSymbolToBeInitialized(member)); [](http://example.com/codeflow?start=40&length=51)

I am curious why are we digging only one level deep. Shouldn't we add NotNullMembers for this property as well?

Reply:
I don't believe that relationship can be implied. All `MemberNotNull` says is that the other member will be not null, not that it will actually call the setter. It could just set a backing field directly, for example, and not run the body which would have had other side effects. So I intentionally worded the spec change to only be one level deep, which is consistent with the behavior of `MemberNotNull` in general.